gc/base/IndexableObjectScanner.hpp

```diff
@@ -91,7 +108,25 @@ class GC_IndexableObjectScanner : public GC_ObjectScanner
 		GC_ObjectScanner::initialize(env);
 	}
 
-public:
+	/**
+	 * Get the current element of the array
+	 */
+	MMINLINE fomrobject_t *getScanPtr() { return _scanPtr; }
+
+	/**
+	 * Get the address of the next element in the array.
+	 * @return NULL if there are no more elements
+	 */
+	MMINLINE fomrobject_t *
+	next()
+	{
+		_scanPtr = (fomrobject_t*)((uintptr_t)_scanPtr + _elementSize);
+		if (_scanPtr < _endPtr) {
+			return _scanPtr;
+		}
+		return NULL;
+	}
```

Comment:
Style preference (you can keep it "as is" - up to you). I would use single return style:
```
       MMINLINE fomrobject_t *
  	next()
  	{
               fomrobject_t *result = NULL;
  		_scanPtr = (fomrobject_t*)((uintptr_t)_scanPtr + _elementSize);
  		if (_scanPtr < _endPtr) {
 			result = _scanPtr;
 		}
 		return result;
 	}
```